### PR TITLE
MB-1809 Left Nav

### PR DESCRIPTION
**Pull request by Shkeating**
## Description

This branch establishes the basic markup and styling for the **Left Nav** component. It also establishes a new type of Tag, `usa-tag--teal` which is included in this component. This element is used in a sidebar, and includes an example of how to use it in a page layout. It also has an example of smooth scrolling to an achor point from one of the nav links.

## Reviewer Notes

Right now, I have smooth scrolling set up by using the css `scroll-behavior: smooth;` which does not have browser support in Safari 😪, so we will need to figure out either a fallback or a different approach in js.

## Setup

MAke sure Docker is running

```sh
yarn install
make storybook
```

## Code Review Verification Steps
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

- [Jira story](https://dp3.atlassian.net/browse/MB-1809?atlOrigin=eyJpIjoiODNhYjI5M2YxNmRjNGUzNjkxYWJlMmJmM2VjMDgyYjIiLCJwIjoiaiJ9) for this change
- [Abstract Reference](https://share.goabstract.com/604f7f29-c8bc-4c91-af36-92cd130b5b3f?collectionLayerId=172b5d4b-30de-4aa9-b0bd-7890cce7ef9f&mode=build) 
- [Axure Prototype of intended scroll behavior](https://xc9rwh.axshare.com/#id=gro6ti&p=move_details_dir_a&g=1)

## Screenshots

leftnav

### CODE REVIEW

File: src/components/LeftNav/index.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import { Tag } from '@trussworks/react-uswds';
+import styles from './index.module.scss';
+import { ReactComponent as AlertIcon } from '../../shared/icon/alert.svg';
+
+const cx = classNames.bind(styles);
+
+const LeftNav = () => (
```
Comment (gidjin) [inline src/components/LeftNav/index.jsx:9]: I'm curious why you didn't / couldn't use the `SideNav` from `react-uswds`. Is our `LeftNav` significantly different?

Comment (Shkeating) [reply src/components/LeftNav/index.jsx:9]: NOPE I'm JUST DUMB AND DIDN'T NOTICE IT EXISTED 😩 taking a look at it now

Comment (Shkeating) [reply src/components/LeftNav/index.jsx:9]: okay so i just looked at it, it still would have needed some pretty substantial customization to look like this one. and the one i built is customized to have a wrapping div to facilitate the fixed position, scroll down the page with the user behavior which is not native to the `SideNav` as far as I can see. Ours also has to have it possible to embed `Tag`s

Comment (gidjin) [reply src/components/LeftNav/index.jsx:9]: Cool, I think we could have found a way to embed the Tag, but it sounds like there was enough other stuff for our own. Thanks for looking.

**Comment by Shkeating** [comment]
> I believe the exclamation icon should use the `warning` yellow. Also, the screenshot's a little blurry but it looks like the blue outline on the left has a 0px radius all around. It should inherit the nav container's radius, so top left corner should be 4px. But ignore this if I'm just seeing things!
> 
> Everything else looks good to me.

@amanda5iu you make a really good point here! what it looks like is happening is that the `border-left` that is applied, makes the corner sharp. Here's the problem though, if I give that border a `border-radius` value, it rounds it at the top like we want, but it also rounds it at the bottom 😵 

i did some internet exploring and learned that there is such thing as indivual `border-radius` attributes for every corner! so I have applied this fancy css that accounts for as many nav items as we want in an indivual component instance ⭐️
```
nav a:first-child:active  {
      border-top-left-radius: 4px;
}
nav a:last-child:active {
      border-bottom-left-radius: 4px;
}
```

BEFORE
image

AFTER
image

**Comment by amanda5iu** [comment]
@Shkeating This looks great! Ship it 🚀

**Pull request merged**